rapthor/lib/operation.py

```diff
@@ -203,19 +207,28 @@ def call_toil(self):
         args.extend(['--writeLogs', self.log_dir])
         args.extend(['--logLevel', 'DEBUG'])
         args.extend(['--preserve-entire-environment'])
-        # Note: instead of using '--preserve-entire-environment', we can pass a limited
-        # number of env variables with something like:
-        # args.extend(['--preserve-environment', 'PATH', 'PYTHONPATH', 'LD_LIBRARY_PATH',
-        #              'MPI_PREFIX', 'LOFARROOT'])
-        # This would avoid passing unwanted variables (e.g., SLURM ones), but we have to
-        # be sure we get all the ones we need, of course
+#         args.extend(['--preserve-environment', 'PATH', 'PYTHONPATH', 'LD_LIBRARY_PATH'])
         if self.scratch_dir is not None:
-            args.extend(['--tmpdir-prefix', self.scratch_dir])
-            args.extend(['--tmp-outdir-prefix', self.scratch_dir])
+            # Note: the trailing '/' is expected by Toil v5.3+
+            args.extend(['--tmpdir-prefix', self.scratch_dir+'/'])
+            args.extend(['--tmp-outdir-prefix', self.scratch_dir+'/'])
         args.extend(['--clean', 'never'])
         args.extend(['--cleanWorkDir', 'never'])
         args.extend(['--servicePollingInterval', '10'])
         args.extend(['--stats'])
+        if self.field.use_mpi:
+            # Create the config file for MPI jobs and add the required args
+            if self.batch_system == 'slurm':
+                # Use salloc to request the SLRUM allocation and run the MPI job
+                config_lines = ["runner: 'mpi_runner.sh'", "nproc_flag: '-N'",
+                                "extra_flags: ['mpirun', '--map-by node']"]
+            else:
```

Comment:
Was this tested on a non-slurm cluster?

Reply:
In line 98 of this file, there is still a reference to 'singleMachine' in the old form

Reply:
No. I will add a warning that it is experimental.